core/impl/src/main/java/com/blazebit/persistence/impl/JoinManager.java

```diff
@@ -691,8 +699,12 @@ private static boolean isNoJoinValuesNode(JoinNode rootNode) {
                 } else {
                     if (type instanceof EntityType<?>) {
                         sb.append(((EntityType) type).getName());
-                        if (rootNode.getValuesIdName() != null) {
-                            sb.append('.').append(rootNode.getValuesIdName());
+
+
+                        Set<String> valuesIdNames = rootNode.getValuesIdNames();
+                        if (valuesIdNames != null && ! valuesIdNames.isEmpty()) {
+                            String next = valuesIdNames.iterator().next();
+                            sb.append(".").append(next);
```

Comment:
I suppose that this has to be altered, but how can one hit this code?

Reply:
When you use `fromIdentifiableValues()` and the generate the _logical query string_ via `CriteriaBuilder.getQueryString()`